src/index.test.js

```diff
@@ -5,12 +5,27 @@ describe("SkynetClient", () => {
     const client = new SkynetClient();
 
     // Download
-    expect(client).toHaveProperty("download");
-    expect(client).toHaveProperty("getDownloadUrl");
-    expect(client).toHaveProperty("open");
+    expect(client).toHaveProperty("downloadFile");
+    expect(client).toHaveProperty("downloadFileHns");
+    expect(client).toHaveProperty("getHnsUrl");
+    expect(client).toHaveProperty("getHnsresUrl");
+    expect(client).toHaveProperty("getSkylinkUrl");
+    expect(client).toHaveProperty("metadata");
+    expect(client).toHaveProperty("openFile");
+    expect(client).toHaveProperty("openFileHns");
+    expect(client).toHaveProperty("resolveHns");
+
+    // Encryption
+    expect(client).toHaveProperty("addSkykey");
+    expect(client).toHaveProperty("createSkykey");
+    expect(client).toHaveProperty("getSkykeyById");
+    expect(client).toHaveProperty("getSkykeyByName");
+    expect(client).toHaveProperty("getSkykeys");
 
     // Upload
-    expect(client).toHaveProperty("upload");
+    expect(client).toHaveProperty("uploadFile");
+    expect(client).toHaveProperty("uploadFileRequest");
     expect(client).toHaveProperty("uploadDirectory");
+    expect(client).toHaveProperty("uploadDirectoryRequest");
```

Comment:
It's a usability impediment to have to append `.skylink` to every upload call. Since that is the desired result 99% of the time we have this as the main API as well a version of the method that returns the full request (with merkleroot and bitfield).